### PR TITLE
shapely 2.0.6

### PR DESCRIPTION
**Pull request by lorepirri**
shapely 2.0.6

**Destination channel:** defaults

### Links

- [PKG-6063]
- dev_url (pypi): https://github.com/shapely/shapely/tree/2.0.6
- diff: https://github.com/shapely/shapely/compare/2.0.5...2.0.6
- conda-forge: https://github.com/conda-forge/shapely-feedstock/blob/main/recipe/meta.yaml
- pypi: https://pypi.org/project/shapely/2.0.6
- pypi inspector: https://inspector.pypi.io/project/shapely/2.0.6

### Depends on 

- AnacondaRecipes/geos-feedstock#11

### Explanation of changes:

- ⚠️ do not merge before deploying `geos` and update the `conda_build_config.yaml` with the last version.
- bump version

[PKG-6063]: https://anaconda.atlassian.net/browse/PKG-6063?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
   name: {{ name|lower }}
   version: {{ version }}
 
 source:
   url: https://github.com/{{ name }}/{{ name }}/archive/refs/tags/{{ version }}.tar.gz
-  sha256: 56c0aee11b31a9dd18120423d350fa6109663d4e46bccc58550bc7d735f18005
+  sha256: 71f1830bd1aa42d0493c8beb48c2e80712450af43376af4ba3bf40f0d9064120
 
 build:
   number: 0
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:13]: Should `numpy>=1.14,<3` be in `run` to support **numpy 2**? https://github.com/shapely/shapely/blob/2.0.6/pyproject.toml#L46C5-L46C19

Comment (lorepirri) [reply recipe/meta.yaml:13]: true, thanks! Updated

**Comment by skupr-anaconda** [comment]
An error:
```
A module that was compiled using NumPy 1.x cannot be run in
NumPy 2.1.3 as it may crash. To support both 1.x and 2.x
versions of NumPy, modules must be compiled with NumPy 2.0.
Some module may need to rebuild instead e.g. with 'pybind11>=2.12'.

If you are a user of the module, the easiest solution will be to
downgrade to 'numpy<2' or try to upgrade the affected module.
We expect that some modules will need time to support NumPy 2.
```
Could you set `numpy 2` in `host`?

**Review by skupr-anaconda — APPROVED** [review]

File: abs.yaml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+channels:
+  - https://staging.continuum.io/prefect/fs/geos-feedstock/pr11/b86b987
```
Comment (skupr-anaconda) [inline abs.yaml]: Can you build it for py313 too?
```suggestion
  - https://staging.continuum.io/prefect/fs/geos-feedstock/pr11/b86b987
  
  build_env_vars:
  ANACONDA_ROCKET_ENABLE_PY313 : yes
  
```

Comment (lorepirri) [reply abs.yaml]: I was just going to add this, thanks!

**Comment by ifitchet** [comment]
Given that we know that previous versions all passed their own tests and it was only `fiona` that tripped over something, do you think we should add that specific test here as a regression test?

The guts of the test is (from https://github.com/Toblerity/Fiona/blob/main/tests/test_features.py#L59-L69):

```python
import sys

import shapely  # type: ignore
from shapely.geometry import LineString, MultiPoint, Point, mapping, shape  # type: ignore

from fiona.features import (  # type: ignore
    vertex_count,
)

obj = Point(0, 0).buffer(10.0).difference(Point(0, 0).buffer(1.0))
count = 130

vc_count = vertex_count(obj)
if not count == vc_count:
    print(f"FAIL {count} != {vc_count}")
    sys.exit(1)
```

**Comment by lorepirri** [comment]
> Given that we know that previous versions all passed their own tests and it was only `fiona` that tripped over something, do you think we should add that specific test here as a regression test?
> 
> The guts of the test is (from https://github.com/Toblerity/Fiona/blob/main/tests/test_features.py#L59-L69):
> 
> ```python
> import sys
> 
> import shapely  # type: ignore
> from shapely.geometry import LineString, MultiPoint, Point, mapping, shape  # type: ignore
> 
> from fiona.features import (  # type: ignore
>     vertex_count,
> )
> 
> obj = Point(0, 0).buffer(10.0).difference(Point(0, 0).buffer(1.0))
> count = 130
> 
> vc_count = vertex_count(obj)
> if not count == vc_count:
>     print(f"FAIL {count} != {vc_count}")
>     sys.exit(1)
> ```

@ifitchet great idea, test added!

**Review by ifitchet — APPROVED** [review]
nit: remove `/en/stable` from `doc_url`

**Pull request merged**